examples/ddi_seal/dataloader.py

```diff
@@ -0,0 +1,115 @@
+import torch
+from torch_geometric.data import InMemoryDataset, Dataset
+from utils import get_pos_neg_edges, extract_enclosing_subgraphs, k_hop_subgraph, construct_pyg_graph
+from torch_sparse import coalesce
+import scipy.sparse as ssp
+
+
+class SEALDataset(InMemoryDataset):
+    def __init__(self, root, data, split_edge, num_hops, percent=100, split='train',
+                 use_coalesce=False, node_label='drnl', ratio_per_hop=1.0,
+                 max_nodes_per_hop=None):
+        self.data = data
+        self.split_edge = split_edge
+        self.num_hops = num_hops
+        self.percent = int(percent) if percent >= 1.0 else percent
+        self.split = split
+        self.use_coalesce = use_coalesce
+        self.node_label = node_label
+        self.ratio_per_hop = ratio_per_hop
+        if not max_nodes_per_hop:
+            max_nodes_per_hop = None
+        self.max_nodes_per_hop = max_nodes_per_hop
+        super(SEALDataset, self).__init__(root)
+        self.data, self.slices = torch.load(self.processed_paths[0])
+
+    @property
+    def processed_file_names(self):
+        if self.percent == 100:
+            name = 'SEAL_{}_data'.format(self.split)
+        else:
+            name = 'SEAL_{}_data_{}'.format(self.split, self.percent)
+        name += '.pt'
+        return [name]
+
+    def process(self):
+        pos_edge, neg_edge = get_pos_neg_edges(self.split, self.split_edge,
+                                               self.data.edge_index,
+                                               self.data.num_nodes,
+                                               self.percent)
+
+        if self.use_coalesce:  # compress mutli-edge into edge with weight
+            self.data.edge_index, self.data.edge_weight = coalesce(
+                self.data.edge_index, self.data.edge_weight,
+                self.data.num_nodes, self.data.num_nodes)
+
+        if 'edge_weight' in self.data:
+            edge_weight = self.data.edge_weight.view(-1)
+        else:
+            edge_weight = torch.ones(self.data.edge_index.size(1), dtype=int)
+        A = ssp.csr_matrix(
+            (edge_weight, (self.data.edge_index[0], self.data.edge_index[1])),
+            shape=(self.data.num_nodes, self.data.num_nodes)
+        )
```

Comment:
A for the affinity matrix? If yes, a more readable name will be better, e.g. affinity_mat